### ISSUE TITLE
go to next change (]g) does not wrap around file

### ISSUE THREAD

**Issue by Ha5hBr0wn** [helix-editor/helix#7209]
### Summary

By default, other goto style commands wrap around the file when they reach the end. This is not happending with go to next change, and it just stays on the last one when you reach it. 

### Reproduction Steps

In a file in a git repo create a couple change. Then cycle through them with ']g'

### Platform

maxOS ssh'ed into Linux

### Terminal Emulator

iTerm

### Helix Version

23.05

**Comment by gabydd** [comment]
This is actually the case for all the unimpaired(``goto_next_``\``goto_prev_``), none of the commands wrap at stop when you get to the end. There doesn't seem to be any issues about it not wrapping so I am assuming this is the desired behavior? I agree it would probably be nice to be able to wrap to the first change one thing you can currently do is use ``[G`` to go to first change and then start going to next changes again.

**Comment by t-monaghan** [comment]
First time contributor here! Please find my first attempt above.

I'm curious, will this require a "wrapped around document" status for the editor? This is how search behaves, but not how diagnostic's next/previous behaves. 

Also, wrapping a document with search seems optional, whereas this optionality is not found with diagnostic gotos, is this intended behaviour?

The way I have written the node captures for the first/last elements also seems less than elegant, and I would be keen to hear any advice on a different approach of finding these nodes.

I'm eager to handle this solution, however I may require guidance due to my inexperience in open-source contributing.

**Comment by the-mikedavis** [comment]
Yeah I think we prefer the current behavior where it doesn't wrap. You can use `[G` as @gabydd says to jump back to the beginning or `]G` to go to the end. It would probably be useful to have a statusline message that says when there are no more matches remaining like when you have `editor.search.wrap-around` set to `false`.
